src/realm/sync/noinst/server/server.cpp

```diff
@@ -1806,16 +1832,17 @@ class HTTPConnection {
                 return;
             }
             negotiated_protocol_version = best_match;
-            logger.debug("Received: Sync HTTP request (negotiated_protocol_version=%1)",
-                         negotiated_protocol_version); // Throws
+            negotiated_protocol_prefix = best_match_prefix;
+            logger.debug("Received: Sync HTTP request (negotiated_protocol_version=%1, prefix='%2')",
+                         negotiated_protocol_version, negotiated_protocol_prefix); // Throws
             formatter.reset();
         }
 
         std::string sec_websocket_protocol_2;
         {
             std::ostringstream out;
             out.imbue(std::locale::classic());
```

Comment:
Instead of all the changes, I would just check `negotiated_protocol_version` against v8:
```
negotiated_protocol_prefix = negotiated_protocol_version >= 8 ? get_pbs_websocket_protocol_prefix() : get_old_pbs_websocket_protocol_prefix();
```